app.js

```diff
@@ -1,63 +1,231 @@
+    } else {
+      log.trace(logData, 'loginSuperUser success');
+    }
+    cb(err);
+  });
+};
+
+/**
+ * Validate query parameters
+ */
+app._validateRequest = function (req, res, next) {
+  log.info('app._validateRequest');
+  if (!req.query.shortHash && req.query.type !== 'signin') {
+    log.trace('_validateRequest !shortHash');
+    // TODO?: switch to createAndReport
+    return next(ErrorCat.create(500, 'instance shortHash required'));
+  }
+  if (!~validDetentionTypes.indexOf(req.query.type)) {
+    // only valid occurance if login error
+    log.trace('_validateRequest !type');
+    // TODO?: switch to createAndReport
+    return next(ErrorCat.create(500, 'invalid request type'));
+  }
+  log.trace('_validateRequest success');
+  next();
 };
 
+/**
+ * Fetch Instance resource from API
+ */
+app._fetchInstance = function (req, res, next) {
+  log.info({
+    shortHash: req.query.shortHash
+  }, 'api._fetchInstance');
+  if (req.query.type === 'signin') {
+    log.trace('_fetchInstance signin bypass');
+    return next();
+  }
+  req.instance = superUser.fetchInstance(req.query.shortHash, function (err) {
+    if (err) {
+      log.error({
+        err: err
+      }, '_fetchInstance superUser.fetchInstance error');
+      // TODO?: switch to createAndReport
+      return next(ErrorCat.create(404, 'instance not found'));
+    }
+    log.trace('_fetchInstance superUser.fetchInstance success');
+    next();
+  });
+};
 // uncomment after placing your favicon in /public
-app.use(bodyParser.json());
-app.use(bodyParser.urlencoded({ extended: false }));
-app.use(express.static(path.join(__dirname, 'public')));
 
-app.route('/*').get(function (req, res, next) {
-  var options = {
-    localVersion: version,
-    absoluteUrl: process.env.ABSOLUTE_URL || 'detention.runnable.io'
-  };
-  if (req.query.type) {
-    var page = req.query.type;
-    [
-      'status',
-      'branchName',
-      'redirectUrl',
-      'containerUrl',
-      'ownerName',
-      'instanceName'
-    ].forEach(function (option) {
-      options[option] = req.query[option];
-    });
-    if (req.query.ports) {
-      var value = req.query.ports;
-      if (!Array.isArray(value)) {
-        value = [value];
-      }
-      options.ports = value;
+/**
+ * Resolve instance status and render + return relevant error message html page
+ */
+app._processNaviError = function (req, res, next) {
+  log.info({
+    query: req.query
+  }, 'processNaviError');
+  var options = {};
+
+  [
+    'redirectUrl',
+    'shortHash'
+  ].forEach(function (option) {
+    options[option] = req.query[option];
+  });
+
+  if (req.instance) {
+    options.branchName = keypather.get(req.instance, 'attrs.contextVersion.branch');
+    // Temp missing pending resolution of SAN-3018
+    // https://runnable.atlassian.net/browse/SAN-3018
+    options.instanceName = keypather.get(req.instance, 'attrs.lowerName');
+    options.ownerName = keypather.get(req.instance, 'attrs.owner.username');
```

Comment:
Yes @Nathan219 this JIRA ticket is tracking that.
https://runnable.atlassian.net/browse/SAN-3018

Not a blocker for deploying the refactored detention as is.